include/kokkos-utils/callbacks/Events.hpp

```diff
@@ -276,7 +276,24 @@ concept EnqueuedEvent = Event<EventType> && requires (EventType event) {
 //! Concept to constrain any event type that is one of the given event types.
 template <typename T, typename... EventTypes>
 concept EventOneOf = impl::type_list_contains_v<T, EventTypes...>;
+///@}
+
+//! @name Get the paired event.
+///@{
+template <BeginEvent EventType>
+struct PairedEventType;
 
+#define PAIRED_EVENT_TYPE(__begin_eventtype__, __end_eventtype__) \
+    template <> \
+    struct PairedEventType<__begin_eventtype__> { using type = __end_eventtype__; };
+
+PAIRED_EVENT_TYPE(BeginParallelForEvent,    EndParallelForEvent)
+PAIRED_EVENT_TYPE(BeginParallelReduceEvent, EndParallelReduceEvent)
+PAIRED_EVENT_TYPE(BeginParallelScanEvent,   EndParallelScanEvent)
+PAIRED_EVENT_TYPE(BeginFenceEvent,          EndFenceEvent)
```

Comment:
Here, I've limited the pairs to the pairs that we're currently using with `BeginEndEventIdMatcher`.

Reply:
I wouldn't do that. I've explicitly avoided the name `BeginEndPair` to allow `PairedEvent` to embrace a broader goal.